lib/archethic/contracts/worker.ex

```diff
@@ -218,18 +223,26 @@ defmodule Archethic.Contracts.Worker do
       contract_transaction = Constants.to_transaction(contract_constants)
 
       if Conditions.empty?(oracle_condition) do
-        with next_tx <- ActionInterpreter.execute(Map.fetch!(triggers, :oracle), constants),
+        with next_tx = %Transaction{} <-
+               ActionInterpreter.execute(Map.fetch!(triggers, :oracle), constants),
              {:ok, next_tx} <- chain_transaction(next_tx, contract_transaction),
              :ok <- ensure_enough_funds(next_tx, address) do
           handle_new_transaction(next_tx)
         end
       else
         with true <- ConditionInterpreter.valid_conditions?(oracle_condition, constants),
-             next_tx <- ActionInterpreter.execute(Map.fetch!(triggers, :oracle), constants),
+             next_tx = %Transaction{} <-
+               ActionInterpreter.execute(Map.fetch!(triggers, :oracle), constants),
              {:ok, next_tx} <- chain_transaction(next_tx, contract_transaction),
              :ok <- ensure_enough_funds(next_tx, address) do
           handle_new_transaction(next_tx)
         else
+          nil ->
+            # returned by ActionInterpreter.execute when contract did not create a next tx
+            Logger.error("Oracle contract did not trigger a new tx",
```

Comment:
Why should it be an error ?

Reply:
Indeed it should not be an error as it may be expected. I alos think that the log error when the conditions are not valids should not be an error also, but just an information.